app/bundles/LeadBundle/Entity/TagRepository.php

```diff
@@ -91,18 +89,38 @@ public function checkLeadByTags(Lead $lead, $tags)
 
         $q = $this->_em->getConnection()->createQueryBuilder();
         $q->select('l.id')
-            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l');
-        $q->join('l', MAUTIC_TABLE_PREFIX.'lead_tags_xref', 'x', 'l.id = x.lead_id')
-                ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags', 't', 'x.tag_id = t.id')
-                ->where(
-                    $q->expr()->andX(
-                        $q->expr()->in('t.tag', ':tags'),
-                        $q->expr()->eq('l.id', ':leadId')
-                    )
+            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l')
```

Comment:
Why do we have here this code with prefix and table names? We can simply use entity class name instead.
Same for joins.

Reply:
@Maxell92 that only works when using DQL. This is a DBAL query so using native SQL here.

Reply:
Ok, so I modify my question:
1) Why is used DBAL and not DQL here? I see no reason for it.
2) I think this method should go inside LeadRepository and not Tag repository. It is selecting from `lead` table, which suggests this.

Reply:
The intention of this PR was not to rewrite the tag functionality. It is adding API endpoint for tags. Although I improved some code while in it. The change in this method was just an indentation fix. I think we are getting far from objective of this PR. We could end up rewriting the whole Mautic like this and the PR will never be done. ;)

Reply:
Leads is a hybrid of ORM and DBAL due to the use of custom fields that manipulate the schema dynamically. Therefore, it is often more efficient to deal with Lead data outside of entity to avoid querying for IDs based on filtered data then querying again with DQL/ ORM to hydrate entities